### ISSUE TITLE
Add a TriOviz anaglyph 3d profile

### ISSUE THREAD

**Issue by Duglim** [vpinball/vpinball#846]
While adjusting anaglyph 3D glasses, the TriOviz glasses can't be calibrated very well and produce a lot of ghosting. These glasses use a variant of Magenta/Green glass, that is slightly different from the normal Magenta/Green option. With the new inbuild glasses calibration in live editor, it is not possible to tweak out the slight color mismatch, as the base colors are too different. 

Feature Request:
Please add an own anaglyph profile for TriOviz glasses, that differs from the stock Magenta/Green.

Additional notes:
- The SuperDepth3D.fx shader in ReShare already has an TriOviz option, maybe this could help.
- https://github.com/vpinball/vpinball/issues/709 is an old issue about this, but the former solution is not suficcient.
- Wikipedia https://en.wikipedia.org/wiki/Anaglyph_3D#Inficolor_3D 

Due to the very good color and light preservation of TriOviz, these glasses seem to be an excellent choice for anaglyph 3D pinball. 

Thanks, 
Duglim

**Comment by Duglim** [comment]
I'm about to pull this request back. For testing purpose I installed Reshade mit 3D Depth Plugin and did setup the inbuild Trioviz options. 
While I was able to minimize ghosting here, this comes at a hugh reduction of the 3D effect. With an similar 3D effect like the inbuild VPX stereo magenta/green option and a correct setup perspective and eye separation of 64mm, the ghosting is even worse with Reshade. Maybe because it's only able to do Fake3D and the inbuild VPS GL solution is real rendering of 2 eyes. 

So all in all, I think the inbuild Anaglyph 3D solution with GL is far superior. 

Maybe there could be some additional settings in the future to better adjust convergence or Max separation height, since the Ghosting maily appears on the very high objects on a table. 
We will see. What I've read, @vbousquet will add nice and great things for Stereo 3D in the future. 

Thx for all the fantastic work here!
Duglim

**Comment by vbousquet** [comment]
I have recently received some Trioviz and I'm testing them at the moment. These are very special glasses since the filters are not mutually exclusive (which is the main property for anaglyph to work without ghosting). So, I suppose they require special image treatment to be used. This will come at a cost (meaning loosing colors / brightness or black) but should be done before judging them too bad

**Comment by Duglim** [comment]
Okay, great to hear. If you received these glasses, I'm happy to wait. I did only read about them and they took 2 weeks to ship from USA to Europe. So no problem waiting some more time. :)

**Comment by vbousquet** [comment]
I think they work fairly differently than traditional anaglyph. So far my understanding is that they are not designed to split the image between the eyes like anaglyph (I would say they are not anaglyph per see, since both eyes see mostly the same image) but to add a depth information using a defocused color channel. Therefore, you actually needs 3 renders:
- the main image (full color) which is the one you look at
- 2 defocused depth render that add the depth information by being merged as green and magenta (without too much perturbating the main render)

This seems to work very well for content designed with that in mind:
- contrasted scenes
- not too much depth budget
- simple alpha test transparency

I think, it will be very difficult to get it right for something like VPX which does not meet this needs. Using them as analgyph glasses seems unlikely to just work (they do not seem to be made for that). Beside the need for 3 renders would be a big change for VPX (and a performance killer). So I won't implement this and I owuld suggest to stick to Reshade (which is likely the right way to generate the depth defocused planes).

**Comment by Duglim** [comment]
I understand. 
From your testing side, are there any glasses you could recommend? 
Red/Blue works fine, but the color and brightness is really low. Are there better color combinations you testes? 
I've read a lot about Amber/Blue beeing much to dark, while the colors are better.
What abour normal Magenta/Green glasses? Was not able to find any glasses to buy...

**Comment by Duglim** [comment]
> I think these might be the Green/Magenta that could actually work with VPX Possibly... https://www.berezin.com/shop/542GM These durable glasses contain optical quality cast acrylic lenses tinted to exacting color standards. The lenses are chromatically tuned to phosphors in computer screens to virtually eliminate ghost images and other annoying visual distortions. And these may be the ones for Red/Blue: https://www.berezin.com/shop/542

Thanks, I ordered 2 and will report on the quality, once they are shipped to germany. :)

**Comment by shro2023** [comment]
> I think they work fairly differently than traditional anaglyph. So far my understanding is that they are not designed to split the image between the eyes like anaglyph (I would say they are not anaglyph per see, since both eyes see mostly the same image) but to add a depth information using a defocused color channel. Therefore, you actually needs 3 renders:
> 
> * the main image (full color) which is the one you look at
> * 2 defocused depth render that add the depth information by being merged as green and magenta (without too much perturbating the main render)
> 
> This seems to work very well for content designed with that in mind:
> 
> * contrasted scenes
> * not too much depth budget
> * simple alpha test transparency
> 
> I think, it will be very difficult to get it right for something like VPX which does not meet this needs. Using them as analgyph glasses seems unlikely to just work (they do not seem to be made for that). Beside the need for 3 renders would be a big change for VPX (and a performance killer). So I won't implement this and I owuld suggest to stick to Reshade (which is likely the right way to generate the depth defocused planes).

Just wanted to add the superdepth3d has TrioViz within its shader profile, would it not be possible to implement in the same way that this has as like the OP being able to use TrioViz would be a big benefit as does seem superior from a colour perspective to the red/blue (which loses all red color) and blue/amber which tends to be very desaturated?

**Comment by Pinbalbilly** [comment]
@Duglim Have you tried the magenta green glasses you ordered?  I'm using the cyan red ones, but I think the cyan is too blue and I can't calibrate them quite well, some ghosting that I can't correct.  Thank you.

**Comment by Duglim** [comment]
> @Duglim Have you tried the magenta green glasses you ordered? I'm using the cyan red ones, but I think the cyan is too blue and I can't calibrate them quite well, some ghosting that I can't correct. Thank you.

Yes, I tried. Sadly, these magenta/green glasses did not match the filter at all. Very much ghosting on both eyes.

**Comment by 0ldG1t00** [comment]
Just wanted to know now that we are almost 2 years since the last post have we gotten anywere more forward with the 3D glasses? Like above posts I have the trioviz, plus another pair of magenta / green and various red and cyan ones. The trioviz certainly seem to have better colour across the board (some tables do look better with red / cyan glasses. As for Reshade, I jsut could not get decent depth with the trioviz glasses despite various attempts and decided that VPX native 3D capabiltites are far superior that Reshade which is why its a shame as we seem to be almost there, just need a couple of fine tune settings?
Is there going to be any more options added for 3D glasses in VPX or are more people shifting over to VR?
